tools/greptimedbcluster/clean-cluster/clean-cluster.sh

```diff
@@ -0,0 +1,149 @@
+                break
+            fi
+            echo "Waiting for PVC '$PVC' to be deleted..."
+        done
+
+        echo -e "${BLUE}Successfully deleted PVC '$PVC'.${RESET}"
+    done
+fi
+
+if [[ -z "$ETCD_NAME" || -z "$ETCD_NAMESPACE" ]]; then
+    exit 0
+fi
+
+# Clean up etcd data
+echo -e "${YELLOW}Starting etcd data cleanup...${RESET}"
+
+# Get etcd StatefulSet pods
+ETCD_SELECTOR=$(kubectl get statefulset -n "$ETCD_NAMESPACE" "$ETCD_NAME" -o jsonpath='{.spec.selector.matchLabels}' | jq -r 'to_entries|map("\(.key)=\(.value|tostring)")|join(",")')
+
+if [ -z "$ETCD_SELECTOR" ]; then
+    echo -e "${RED}Failed to get ETCD StatefulSet ${ETCD_NAME} in namespace ${ETCD_NAMESPACE}${RESET}"
+    exit 1
+fi
+
+# Get only Running etcd pods using the StatefulSet's selector
+ETCD_PODS=$(kubectl get pods -n "$ETCD_NAMESPACE" -l "$ETCD_SELECTOR" --field-selector status.phase=Running -o jsonpath='{.items[*].metadata.name}')
+
+if [ -z "$ETCD_PODS" ]; then
+    echo -e "${RED}No Running etcd pods found for StatefulSet '${ETCD_NAME}' in namespace '${ETCD_NAMESPACE}'.${RESET}"
+    exit 1
+fi
+
+detect_etcd_leader() {
+    local leader=""
+    for pod in $ETCD_PODS; do
+        echo -e "${CYAN}Checking pod $pod for leader status...${RESET}" >&2
+        local endpoint_status=$(kubectl exec -n "$ETCD_NAMESPACE" "$pod" -- etcdctl endpoint status --write-out=json 2>/dev/null)
+
+        if [ -z "$endpoint_status" ]; then
+            echo -e "${RED}Failed to get status from $pod. Skipping.${RESET}" >&2
+            continue
+        fi
+
+        # Handle both array and single object response formats
+        local member_id=$(echo "$endpoint_status" | jq -r 'if type == "array" then .[0].Status.header.member_id else .Status.header.member_id end')
+        local leader_id=$(echo "$endpoint_status" | jq -r 'if type == "array" then .[0].Status.leader else .Status.leader end')
+
+        if [ -z "$member_id" ] || [ -z "$leader_id" ]; then
+            echo -e "${RED}Invalid status format from pod $pod.${RESET}" >&2
+            continue
+        fi
+
+        if [ "$member_id" = "$leader_id" ]; then
+            leader="$pod"
+            echo -e "${GREEN}Leader found: $pod (member_id=$member_id)${RESET}" >&2
+            break
+        fi
+    done
+
+    if [ -z "$leader" ]; then
+        echo -e "${RED}Error: No etcd leader detected.${RESET}" >&2
+        return 1
+    fi
+    echo "$leader"  # Return just the pod name without any formatting
+}
+ETCD_LEADER=$(detect_etcd_leader)
+
+if [ -z "$ETCD_LEADER" ]; then
+    echo -e "${RED}Failed to detect etcd leader.${RESET}"
+    exit 1
+fi
+
+# Delete all data from etcd
+echo -e "${YELLOW}Deleting all data from etcd...${RESET}"
+if kubectl exec -n "$ETCD_NAMESPACE" "$ETCD_LEADER" -- etcdctl del --prefix ""; then
+    echo -e "${BLUE}Successfully deleted all data from etcd.${RESET}"
+else
+    echo -e "${RED}Failed to delete data from etcd.${RESET}"
+    exit 1
+fi
```

Comment:
Let's add some message here that user still need to manually wipe object storage, RDS and kafka if any